### ISSUE TITLE
Tests failing in lucene_grammar.py example

### ISSUE THREAD

**Issue by lijenicol** [pyparsing/pyparsing#455]
This is nothing too urgent, I am just flagging this - I have noticed that when running the `lucene_grammar.py` example, that some of the failure tests are actually succeeding, which means that either the `expression` or tests aren't correct. If people are using the code from this example to parse lucene queries, then I think it is important that it can be trusted. 

For example, 
parsing this test query that is expected to fail: `a\:b\+c\~` 
yields `['a:b+c~']` (it doesn't fail)

**Comment by ptmcg** [comment]
Thanks for pointing this up, I'll look at restructuring these tests to better detect the failing search strings.

**Comment by ptmcg** [comment]
I wrote this example a few years ago, so I didn't remember that it was already structured into passing and failing tests. But I did go through and review the tests and re-reviewed the Lucene grammar docs.  It looks like some of the fail tests were actually good queries, and vice versa. Also, there was incomplete support for embedded '?' wildcard characters.

I've fixed everything up, and have added this example to the test_examples.py test script, so it will be validated as part of the package testing from here on out.

Thanks for bringing this to my attention!
